Android: Update deps, target SDK 30 and ARMv8

EwoutH opened the pull request:
- Builds for ARMv8 instead of ARMv7
 - Use latest compile SDK

aarongrider commented:
At this point we aren't wanting to update our baseline to Android SDK 30. In fact I actually just downgraded zxing:core to prevent a crashing bug on downlevel devices, but I would be happy to abiFilters change.

EwoutH commented:
I resolved the conflicts with the master branch and set the target back to 28. Compile SDK should stay at 30 since that doesn't change runtime behaviour but does give warning and errors based on the latest Android Studio version.

> It should be emphasized that modifying compileSdkVersion does not change the runtime behavior. When you modify compileSdkVersion, new compilation warnings and errors may appear, but the new compileSdkVersion will not be included in APK: it is only used during compilation. (you really should fix these warnings. There must be a reason for their appearance.)
> 
> Therefore, we strongly recommend that you always compile with the latest SDK. There are many benefits to using new compilation checks on existing code, avoiding new deprecated APIs, and preparing for new APIs.

I also think the versioning is outdated and should either be updated or removed, but maybe that's something for another PR:
```
        versionCode 1
        versionName "1.0"
```

aarongrider commented:
Sounds good. Yes, we should update the version code, good catch. Feel free to either throw it in this PR or a separate, whichever you prefer. Thanks!

EwoutH commented:
Let’s do it separately and merge this first

EwoutH commented:
Solved conflicts, rebased and cleaned up commits. CI passes, this PR has become really small since the dependencies and targetSDK have been rolled back.
